docs/migration-guides/1.11-1.12.md

```diff
@@ -13,4 +13,181 @@ During migration, it may turn out that some resource specifications are too comp
+    ```
+
+3. Edit the Subscriber and add the Owner Reference.
+
+    a. Edit the Trigger:
+
+    ```bash
+    kubectl edit triggers.eventing.knative.dev {TRIGGER_NAME} -n {NAMESPACE}
+    ```
+
+    b. Change the **subscriber** field to:
+
+    ```yaml
+    spec:
+      subscriber:
+        ref:
+          apiVersion: serving.knative.dev/v1
+          kind: Service
+          name: {NAME}
+          namespace: {NAMESPACE}
+    ```
+
+    c. Add the Owner Reference:
+
+    ```yaml
+    metadata:
+      ownerReferences:
+      - apiVersion: serving.knative.dev/v1
+        kind: Service
+        name: {NAME}
+        uid: {FUNCTION_UUID}
+    ```
+
+4. Check if the Trigger is ready:
+
+    ```bash
+    kubectl get triggers.eventing.knative.dev {TRIGGER_NAME} -n {NAMESPACE}
+    ```
+
+### Migrate the APIRule that exposes the Function
+
+1. Find the APIRule with the name starting from the Function name. To list all APIRules in a Namespaces, run this command:
+
+    ```bash
+    kubectl get apirules.gateway.kyma-project.io -n {NAMESPACE} | grep {NAME}-
+    ```
+
+    Check which API Rule points to the Service that has the same name as the Function (**spec.service.name** field).
+
+2. Edit the **handler** and **service** fields in the API Rule:
+
+    ```bash
+    kubectl edit apirules.gateway.kyma-project.io {API_NAME} -n {NAMESPACE}
+    ```
+
+    If the value of **spec.rules.accessStrategies.handler** is `allow`, change it to `noop`. Change the value of the **spec.service.port** field to `80`.
+
+### Migrate the ServiceBindingUsage that binds the Service to the Function
+
+1. Check if the new Function is running:
+
+    ```bash
+    kubectl get functions.serverless.kyma-project.io {NAME} -n {NAMESPACE}
+    ```
+
+2. List ServiceBindingsUsages assigned to the deprecated Function:
+
+    ```bash
+    kubectl get servicebindingusages.servicecatalog.kyma-project.io -l Function={NAME} -n {NAMESPACE}
+    ```
+
+3. Copy the ServiceBindingUsage to this file:
+
+    ```bash
+    kubectl get servicebindingusages.servicecatalog.kyma-project.io {BINDING_NAME} -n {NAMESPACE} -o yaml > binding.yaml
+    ```
+
+4. Delete the deprecated ServiceBindingUsage:
+
+    ```bash
+    kubectl delete servicebindingusages.servicecatalog.kyma-project.io {BINDING_NAME} -n {NAMESPACE}
+    ```
+
+5. Create a new ServiceBindingUsage based on values from the deprecated one (`binding.yaml`):
+
+    ```bash
+    cat <<EOF | kubectl create -f  -
+    apiVersion: servicecatalog.kyma-project.io/v1alpha1
+    kind: ServiceBindingUsage
+    metadata:
+      labels: {metadata.labels}
+      generateName: {FUNCTION_NAME}-
+      namespace: {NAMESPACE}
+      ownerReferences: {metadata.ownerReferences}
+    spec:
+      parameters: {spec.parameters}
+      reprocessRequest: {spec.reprocessRequest}
+      serviceBindingRef: {spec.serviceBindingRef}
+      usedBy:
+        kind: knative-service
+        name: {FUNCTION_NAME}
+    EOF
```

Comment:
I have question about such a sbu: 
```
apiVersion: servicecatalog.kyma-project.io/v1alpha1
kind: ServiceBindingUsage
metadata:
  creationTimestamp: "2020-04-24T13:54:12Z"
  generation: 60
  labels:
    Function: redis
    ServiceBinding: vxzplagkzr
  name: vxzplagkzr
  namespace: aerfio
  ownerReferences:
    - apiVersion: servicecatalog.k8s.io/v1beta1
      kind: ServiceBinding
      name: vxzplagkzr
      uid: f9ecf262-2541-4a64-832d-d69fbaec40c2
  resourceVersion: "144735"
  selfLink: /apis/servicecatalog.kyma-project.io/v1alpha1/namespaces/aerfio/servicebindingusages/vxzplagkzr
  uid: 5bb77591-6e47-43e9-87eb-d5d9a4decf36
spec:
  parameters:
    envPrefix:
      name: REDIS
  reprocessRequest: 3
  serviceBindingRef:
    name: vxzplagkzr
  usedBy:
    kind: function
    name: redis
status:
  conditions:
    - lastTransitionTime: "2020-04-24T14:16:01Z"
      lastUpdateTime: "2020-04-24T15:18:57Z"
      message:
        'Process error during 15 attempts from 15; while ensuring proper labels
        on kind function: while checking if applied labels are equal with current ones:
        while getting injected labels: while listing resources: functions.kubeless.io
        "redis" is forbidden: User "system:serviceaccount:kyma-system:service-catalog-addons-service-binding-usage-controller"
        cannot get resource "functions" in API group "kubeless.io" in the namespace
        "aerfio"'
      reason: EnsureLabelsAppliedError
      status: "False"
      type: Ready

```

From what I understand I can understand the `spec` of the new sbu would look like exactly the same, yeah? I would like to have this clarified, because for now in old sbu you have
```
spec: 
   parameters:
       envPrefix:
            name: REDIS
```
so in this nomenclature  by `parameters: {spec.parameters}` you mean ^ this thing, yeah?

Reply:
yes

Reply:
Hm... can you check if changing `usedBy.kind` will work if you add proper rbac to SBU ctrl? I don't have cluster right now. If it will work, then the instruction will be much simpler

Reply:
Ok, edit works